primitives/core/src/ecdsa.rs

```diff
@@ -537,6 +537,21 @@ impl Pair {
 		let message = secp256k1::Message::parse(message);
 		secp256k1::sign(&message, &self.secret).into()
 	}
+
+	/// Verify a signature on a pre-hashed message. Return `true` if the signature is valid
+	pub fn verify_prehashed(&self, sig: &Signature, message: &[u8; 32], public: &Public) -> bool {
```

Comment:
I would actually move this to the `Signature` type.

Reply:
That is certainly an option.

I choose to add it to the `Pair` type for symmetry. The `blake_256` based `sign()` and `verify()` (non pre-hashed) methods are both defined on `Pair` as well. 

`Signature` does have a non  pre-hashed `recover()` method.  So it could be argued, that `Signature` should have a `recover_prehashed` method as well for being truly symmetric :thinking:

Reply:
If we keep it here, we should drop `&self` at least.

Reply:
Yeah, I'd be in favour of adding `recover_prehashed` to `Signature` and using it here + dropping `&self` as Basti mentioned.

Reply:
Will add `Signature::recover_prehashed()` as well. Not sure, however, what you mean by `and using it here`. `recover()` recovers the public key from the signature :thinking:

Reply:
I was thinking we could `recover_prehashed` and then compare with public key to avoid code duplication, but `full-crypto` feature might complicate some of this stuff :)